File: client/src/types/Message.ts
import type { MessageType } from './enum/MessageType'

export interface Message {
  id: string
  content?: string | null
  senderId?: string
  senderName?: string
  chatId: string
  messageType?: MessageType
  mediaUrl?: string | null
  replyToMessageId?: string | null
  replyToMessageContent?: string | null
  replyToMessageType?: MessageType | null
  replyToMessageMediaUrl?: string | null
  replyToMessageSenderId?: string | null
  isReplyMessageEdited?: boolean
  isReplyMessageDeleted?: boolean
  isEdited?: boolean
  isDeleted?: boolean
  createdAt: string
  updatedAt: string
  readInfo?: MessageReadInfo[]
}

interface MessageReadInfo {
  userId: string
  readAt: string
}

File: server/src/main/java/chitchat/model/Message.java
package chitchat.model;

import chitchat.model.auditing.AbstractAuditingDocument;
import chitchat.model.enumeration.MessageType;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Document(collection = "messages")
public class Message extends AbstractAuditingDocument {

    @Id
    private String id;

    private String content;

    private String senderId;

    private String chatId;

    private MessageType messageType;

    private String mediaUrl;

    private String replyToMessageId;

    private String replyToMessageContent;

    private MessageType replyToMessageType;

    private String replyToMessageMediaUrl;

    private String replyToMessageSenderId;

    private Boolean isReplyMessageEdited;

    private Boolean isReplyMessageDeleted;

    @Builder.Default
    private Boolean isEdited = false;

    @Builder.Default
    private Boolean isDeleted = false;
}
